src/logger.coffee

```diff
@@ -5,114 +5,219 @@ es = require 'event-stream'
 Lock = require 'rwlock'
 { checkTruthy } = require './lib/validation'
 
-LOG_PUBLISH_INTERVAL = 110
+class NativeLoggerBackend
+	constructor: ->
+		@logPublishInterval = 1000
+		@maxLogLinesPerBatch = 10
+		@maxQueueSize = 100
+		@publishQueue = []
+		@intervalHandle = null
+		@publishEnabled = false
+		@offlineMode = false
+
+	doPublish: =>
+		return if @offlineMode or !@publishEnabled or @publishQueue.length is 0
+		# Silently ignore errors sending logs (just like with pubnub)
+		@apiBinder.logBatch(_.slice(@publishQueue, 0, @maxLogLinesPerBatch))
+		.catch(->)
+		@publishQueue = _.slice(@publishQueue, @maxLogLinesPerBatch)
 
-# Pubnub's message size limit is 32KB (unclear on whether it's KB or actually KiB,
-# but we'll be conservative). So we limit a log message to 2 bytes less to account
-# for the [ and ] in the array.
-MAX_LOG_BYTE_SIZE = 30000
-MAX_MESSAGE_INDEX = 9
+	logDependent: (msg, { uuid }) ->
+		if msg.message?.length > 300
+			console.log('Ignored extra-long log line')
+			return
+		@apiBinder.logDependent(uuid, msg)
 
-module.exports = class Logger
-	constructor: ({ @eventTracker }) ->
+	log: (message) =>
+		return if @offlineMode or !@publishEnabled or @publishQueue.length >= @maxQueueSize
+		if _.isString(message)
+			message = { message: message }
+
+		if message.message.length > 300
+			console.log('Ignored extra-long log line')
+			return
```

Comment:
We should truncate log lines instead of dropping them completely

Reply:
Or even better, split the rest of the log message onto a new line

Reply:
Good point, for now I'll just truncate it to 300.

Reply:
Done